atomate/vasp/firetasks/exchange.py

```diff
@@ -0,0 +1,352 @@
+from fireworks import FiretaskBase, FWAction, explicit_serialize
+
+from monty.serialization import loadfn, dumpfn
+
+from atomate.utils.utils import get_logger, env_chk
+from atomate.vasp.database import VaspCalcDb
+
+from datetime import datetime
+import numpy as np
+
+from pymatgen import Structure
+from pymatgen.command_line.bader_caller import bader_analysis_from_path
+from pymatgen.analysis.magnetism.heisenberg import HeisenbergMapper, HeisenbergModel
+from pymatgen.analysis.magnetism import CollinearMagneticStructureAnalyzer, Ordering
+from pymatgen.command_line.vampire_caller import VampireCaller
+
+__author__ = "Nathan C. Frey"
+__email__ = "ncfrey@lbl.gov"
+
+logger = get_logger(__name__)
+
+
+@explicit_serialize
+class HeisenbergModelMapping(FiretaskBase):
+    """
+    Map structures and energies to a Heisenberg model and compute exchange
+    parameters for a given NN cutoff.
+
+    Args:
+        db_file (str): path to file containing the database credentials.
+        wf_uuid (int): Unique id for record keeping.
+        cutoff (float): NN search cutoff (Angstrom).
+        tol (float): distance tolerance for similar bonds.
+        average (bool): <J> only or full NN, NNN HeisenbergModel.
+        structures (list): Magnetic structures.
+        energies (list): Energies / atom (eV).
+
+    """
+
+    required_params = [
+        "db_file",
+        "wf_uuid",
+        "parent_structure",
+        "cutoff",
+        "tol",
+        "average",
+    ]
+
+    # If only doing <J>, give the original structure/energy inputs
+    optional_params = ["structures", "energies"]
+
+    def run_task(self, fw_spec):
+
+        db_file = env_chk(self["db_file"], fw_spec)
+        wf_uuid = self["wf_uuid"]
+        formula = self["parent_structure"].formula
```

Comment:
It seems formula and formula_pretty are never used in this Firetask and parent structure is only required to get the formulas.

Can you remove these lines and also remove parent_structure as a required param.